android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java

```diff
@@ -518,6 +518,8 @@ protected void handleClose(KrollDict options)
 		if (activity != null && !activity.isFinishing() && !activity.isDestroyed()) {
 			activity.finish();
 		}
+
+		// TODO: Does this ever fire the close event?!
```

Comment:
Looks like the `close` event is only fired from `closeFromActivity()`

Reply:
I agree with the current behavior. The "close" event should only be fired once the activity window has been closed.

On Android, we cannot create/destroy activity windows directly. We can only request the OS to do it. The Android OS will not let us create/destroy (ie: open/close a window) while the app is in the background. And the Android OS can destroy an activity window (hence close it) for various reasons such as via back button, intents have particular flags, when Android's "Don't keep activities" is enabled, etc.

The "Don't keep activities" flag is the most interesting one to watch out for. When set, the parent activity window will be temporarily destroyed when displaying a child activity or homing-out, but will be re-created/opened when navigating back.

Honestly, I don't think we should muck with the open/close behavior too much. Perhaps what we really want is a "closing" event (because that's what is being requested) and keep the current "close" event (which I wish was called "closed" past-tense instead).